src/clib/cool1d_multi_g.F

```diff
@@ -731,50 +1145,380 @@ subroutine cool1d_multi_g(
+          
+            call calc_tdust_1d_g(tSiO2D   , tgas, mynh, gasgr_tSiO2D   ,
+     &           gisrfSiO2D   , myisrf, itmask_metal, comp2, 
+     &           in, is, ie, j, k
+     &         , gr_N, gr_Size, gr_dT, gr_Td, alSiO2D   , kpSiO2D    
+     &         , idspecies)
+          
+            call calc_tdust_1d_g(tMgO     , tgas, mynh, gasgr_tMgO     ,
+     &           gisrfMgO     , myisrf, itmask_metal, comp2, 
+     &           in, is, ie, j, k
+     &         , gr_N, gr_Size, gr_dT, gr_Td, alMgO     , kpMgO      
+     &         , idspecies)
+          
+            call calc_tdust_1d_g(tFeS     , tgas, mynh, gasgr_tFeS     ,
+     &           gisrfFeS     , myisrf, itmask_metal, comp2, 
+     &           in, is, ie, j, k
+     &         , gr_N, gr_Size, gr_dT, gr_Td, alFeS     , kpFeS      
+     &         , idspecies)
+          
+            call calc_tdust_1d_g(tAl2O3   , tgas, mynh, gasgr_tAl2O3   ,
+     &           gisrfAl2O3   , myisrf, itmask_metal, comp2, 
+     &           in, is, ie, j, k
+     &         , gr_N, gr_Size, gr_dT, gr_Td, alAl2O3   , kpAl2O3    
+     &         , idspecies)
+            endif
+
+            if (idspecies .gt. 2) then
+            call calc_tdust_1d_g(treforg  , tgas, mynh, gasgr_treforg  ,
+     &           gisrfreforg  , myisrf, itmask_metal, comp2, 
+     &           in, is, ie, j, k
+     &         , gr_N, gr_Size, gr_dT, gr_Td, alreforg  , kpreforg   
+     &         , idspecies)
+          
+            call calc_tdust_1d_g(tvolorg  , tgas, mynh, gasgr_tvolorg  ,
+     &           gisrfvolorg  , myisrf, itmask_metal, comp2, 
+     &           in, is, ie, j, k
+     &         , gr_N, gr_Size, gr_dT, gr_Td, alvolorg  , kpvolorg   
+     &         , idspecies)
+          
+            call calc_tdust_1d_g(tH2Oice  , tgas, mynh, gasgr_tH2Oice  ,
+     &           gisrfH2Oice  , myisrf, itmask_metal, comp2, 
+     &           in, is, ie, j, k
+     &         , gr_N, gr_Size, gr_dT, gr_Td, alH2Oice  , kpH2Oice   
+     &         , idspecies)
+            endif
+
+         endif
+
+!     Calculate dust cooling rate
 
          do i = is+1, ie+1
-            if ( itmask(i) ) then
-               edot(i) = edot(i) - 
+            if ( itmask_metal(i) ) then
+
+            if (idspecies .eq. 0) then
+
+               Ldst(i) =         - 
      &              gasgr(i) * (tgas(i) - tdust(i)) * 
      &              dust2gas(i) * rhoH(i) * rhoH(i)
+
+            else ! idspecies
+
+               if (itdmulti .eq. 0) then
+                   Ldst(i) = - gasgr(i) * (tgas(i) - tdust(i))
+     &                * d(i,j,k) * rhoH(i)
+               else
+
+               if (idspecies .gt. 0) then
+                  Ldst(i) = - (
+     &                gasMgSiO3  (i) * (tgas(i) - tMgSiO3  (i))
+     &              + gasAC      (i) * (tgas(i) - tAC      (i))
+     &              ) * d(i,j,k) * rhoH(i)
+               endif
+
+               if (idspecies .gt. 1) then
+                  Ldst(i) = - (
```

Comment:
Since the `idspecies` are cumulative, I think this needs to add to the `Ldst` value that was calculated just above (1409-1414).
```suggestion
                  Ldst(i) = Ldst(i) - (
```